### PR TITLE
metrics: etcd_object_counts->apiserver_storage_objects

### PR DESCRIPTION
**Pull request by Sallyan**
**How to categorize this PR?**
/area monitoring
/kind enhancement

**What this PR does / why we need it**:
Metric `etcd_object_counts` in etcd3 storage metrics, which is to show number of stored objects. 
And this metric is deprecated in version `1.22.0`, and will be replaced by `apiserver_storage_object_counts`
NOTE: `apiserver_storage_object_counts` is renamed to `apiserver_storage_objects` https://github.com/kubernetes/kubernetes/pull/100082
```
	etcdObjectCounts = compbasemetrics.NewGaugeVec(
		&compbasemetrics.GaugeOpts{
			Name:              "etcd_object_counts",
			DeprecatedVersion: "1.22.0",
			Help:              "Number of stored objects at the time of last check split by kind. This metric is replaced by apiserver_storage_object_counts.",
			StabilityLevel:    compbasemetrics.ALPHA,
		},
		[]string{"resource"},
	)
```
This PR will add `apiserver_storage_objects` to allowed metrics. 

And the metric `etcd_object_counts` is used in three dashboards, if the Kubernetes version is 1.22.0+, this panel will show no data as the metric is deprecated. This PR also updates the dashboards to use two query of the two metrics. 

Reference: https://github.com/kubernetes/kubernetes/issues/98270

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

### CODE REVIEW

**Comment by Sallyan** [comment]
Will check the unit test failure. 
```
> Generate / Vendor Check
>> make generate
>> make revendor
make revendor needs to be run:
 M vendor/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go
make: *** [Makefile:186: check-generate] Error 1
```

File: vendor/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ var (
 	)
 	objectCounts = compbasemetrics.NewGaugeVec(
 		&compbasemetrics.GaugeOpts{
-			Name:           "apiserver_storage_objects",
+			Name:           "apiserver_storage_object_counts",
```
Comment (istvanballok) [inline vendor/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go]: Note that this file is a vendored (hence for us it is a conceptually read-only) copy of the upstream k8s source code, and it is not meaningful to change it.

Generally, if you'd like to change some files in the vendor folder, I think it should be changed upstream, after a discussion with the upstream community.

https://github.com/kubernetes/kubernetes/blob/5f6b60bb1323a60f59119e315ac52b8a453a3939/staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go#L52

Note also that after the initial implementation was announced in the release notes and in the help text you mentioned, the metric name was adjusted in this PR: https://github.com/kubernetes/kubernetes/pull/100082

So please adjust this PR to use the new metric name: `apiserver_storage_objects`.

You can verify it with the following command:

```
k proxy
curl -s http://localhost:8001/metrics | grep apiserver_storage_objects
```

File: pkg/operation/botanist/component/kubeapiserver/monitoring.go [hunk 2]
```diff
@@ -214,6 +215,7 @@ var (
 		monitoringMetricApiserverWatchEventsTotal,
 		monitoringMetricEtcdDbTotalSizeInBytes,
 		monitoringMetricEtcdObjectCounts,
+    monitoringMetricApiserverStorageObjectCounts,
```
Comment (istvanballok) [inline pkg/operation/botanist/component/kubeapiserver/monitoring.go]: inconsistent indentation

**Review by istvanballok — CHANGES_REQUESTED** [review]
with @rickardsjp

File: vendor/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go (hunk 1 above)
Comment (Sallyan) [reply vendor/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go]: Okay, thanks a lot for the information. Will update this PR.

File: pkg/operation/botanist/component/kubeapiserver/monitoring.go (hunk 2 above)
Comment (Sallyan) [reply pkg/operation/botanist/component/kubeapiserver/monitoring.go]: Updated `go fmt`:)

File: vendor/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go (hunk 1 above)
Comment (istvanballok) [reply vendor/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go]: Please make sure to review (and adjust) the PR description as well and revert the change in the vendor folder. For simplicity, you could also squash the commits.

**Review by rfranzke — COMMENTED** [review]
/approve

@wyb1 @istvanballok @rickardsjp please have a look and provide feedback

**Review by istvanballok — APPROVED** [review]
/lgtm

**Comment by rfranzke** [comment]
@acumino Can we add this to the v1.52 milestone?

**Comment by acumino** [comment]
> @acumino Can we add this to the v1.52 milestone?

Yes, but tests are still failing that's why I have not added it to 1.52

**Comment by rfranzke** [comment]
This looks unrelated 👀 

/retest-required

**Comment by oliver-goetz** [comment]
/override check-release-milestone
The status of the last commit does not change anymore, if the milestone was added afterwards.

**Comment by oliver-goetz** [comment]
What a shame, the PR status is still not updated 😓

**Comment by oliver-goetz** [comment]
Awesome 😃

**Comment by vpnachev** [comment]
Was the https://github.com/gardener/gardener/blob/a92a54f2792f7952498fb9473acf5e4d4494dcf8/pkg/operation/botanist/component/etcd/monitoring.go#L154-L155 

skipped by incident or it does not need to be adapted to the new metric name?
cc @istvanballok @wyb1

**Comment by Sallyan** [comment]
> Was the
> 
> https://github.com/gardener/gardener/blob/a92a54f2792f7952498fb9473acf5e4d4494dcf8/pkg/operation/botanist/component/etcd/monitoring.go#L154-L155
> 
> skipped by incident or it does not need to be adapted to the new metric name? cc @istvanballok @wyb1

I think it's better to keep this one and  add new record: 
```
   - record: shoot: apiserver_storage_objects:sum_by_resource 
     expr: max(apiserver_storage_objects) by (resource) 
```

**Pull request merged**